core/node/da_clients/src/avail/client.rs

```diff
@@ -39,37 +137,93 @@ impl DataAvailabilityClient for AvailClient {
+                    AvailClientConfig::FullClient(conf) => conf,
+                    _ => unreachable!(), // validated in protobuf config
+                };
+                let ws_client = WsClientBuilder::default()
+                    .build(default_config.api_node_url.clone().as_str())
+                    .await
+                    .map_err(to_non_retriable_da_error)?;
+
+                let extrinsic = client
+                    .build_extrinsic(&ws_client, data)
+                    .await
+                    .map_err(to_non_retriable_da_error)?;
+
+                let block_hash = client
+                    .submit_extrinsic(&ws_client, extrinsic.as_str())
+                    .await
+                    .map_err(to_non_retriable_da_error)?;
+                let tx_id = client
+                    .get_tx_id(&ws_client, block_hash.as_str(), extrinsic.as_str())
+                    .await
+                    .map_err(to_non_retriable_da_error)?;
+                Ok(DispatchResponse::from(format!("{}:{}", block_hash, tx_id)))
+            }
+            AvailClientMode::GasRelay(client) => {
+                let (block_hash, extrinsic_index) = client
+                    .post_data(data)
+                    .await
+                    .map_err(to_retriable_da_error)?;
+                Ok(DispatchResponse {
+                    blob_id: format!("{:x}:{}", block_hash, extrinsic_index),
+                })
+            }
+        }
     }
 
     async fn get_inclusion_data(
         &self,
-        _blob_id: &str,
+        blob_id: &str,
     ) -> anyhow::Result<Option<InclusionData>, DAError> {
-        // TODO: implement inclusion data retrieval
-        Ok(Some(InclusionData { data: vec![] }))
+        let (block_hash, tx_idx) = blob_id.split_once(':').ok_or_else(|| DAError {
+            error: anyhow!("Invalid blob ID format"),
+            is_retriable: false,
+        })?;
+        let url = format!(
+            "{}/eth/proof/{}?index={}",
+            self.config.bridge_api_url, block_hash, tx_idx
+        );
+        // record current time
+        let current_timestamp = std::time::Instant::now();
+        let bridge_api_data = loop {
+            let response = self
+                .api_client
+                .get(&url)
+                .send()
+                .await
+                .map_err(to_retriable_da_error)?;
+            let bridge_api_data = response
+                .json::<BridgeAPIResponse>()
+                .await
+                .map_err(to_retriable_da_error)?;
+            if bridge_api_data.error.is_none() {
+                break bridge_api_data;
+            }
+            if current_timestamp.elapsed().as_secs() > self.config.timeout as u64 {
+                return Err(to_non_retriable_da_error(anyhow!(
+                    "Inclusion check timeout exceeded"
+                )));
+            }
+            tokio::time::sleep(tokio::time::Duration::from_secs(60)).await;
```

Comment:
Please use [backon](https://docs.rs/backon/latest/backon/) instead of custom loop + check. 
Also, this sleep looks unreasonably long. At least it should be capped by  `config.timeout`

Reply:
I tried `backon` but imo the implementation is much less cleaner because of having to use async closures. Admittedly, maybe I'm just not used to writing good code with `backon`.

Reply:
>Also, this sleep looks unreasonably long. At least it should be capped by config.timeout

what about this?

Reply:
Let me address it without using `backon` for now, I've asked @dimazhornyk to take a look at implementing using backon.

Reply:
> At least it should be capped by config.timeout
Will do.

Reply:
removed the retry logic from here completely as we have it in `da_dispatcher`, so there's no need in the duplication